### PR TITLE
indentity server login with roles

### PR DESCRIPTION
**Pull request by thordurhhh**
login with IDS

* Logs in with ids
* Checks internal DB for matching user for setting role
* If all good, user is logged in.

### CODE REVIEW

File: apps/official-journal-web/src/lib/api/buildFetch.ts [hunk 1]
```diff
@@ -9,15 +9,22 @@ import { EnhancedFetchAPI } from './types'
...
-        // Normalize Request.
+      return async (input, init = {}, token?: string) => {
+        // Normalize Request
         const request = new Request(input as RequestInfo, init as RequestInit)
-        const response = firstMiddleware(request)
+
+        // If a token is provided, add it to the Authorization header
+        if (token) {
+          request.headers.set('Authorization', `Bearer ${token}`)
+        }
```
Comment (thordurhhh) [inline apps/official-journal-web/src/lib/api/buildFetch.ts:24]: Maybe not needed anymore?

File: apps/official-journal-web/src/lib/utils.tsx [hunk 2]
```diff
@@ -388,3 +389,16 @@ export const getOverviewStatusColor = (status: string) => {
       return 'blue'
   }
 }
+
+export const addAuthHeader = (token?: string): Middleware => {
```
Comment (thordurhhh) [inline apps/official-journal-web/src/lib/utils.tsx:393]: Create specific middleware

File: apps/official-journal-web/src/pages/innskraning.tsx [hunk 3]
```diff
@@ -36,8 +30,13 @@ export default function Login(
...
-            <Button onClick={() => handleLogin('0101857799')}>Ármann</Button>
-            <Button onClick={() => handleLogin('0101876689')}>Pálína</Button>
+            <Button
+              onClick={() => signIn(identityServerId, { callbackUrl: '/' })}
+            >
+              IDS
+            </Button>
+            {/* <Button onClick={() => handleLogin('0101857799')}>Ármann</Button>
+            <Button onClick={() => handleLogin('0101876689')}>Pálína</Button> */}
           </Box>
```
Comment (thordurhhh) [inline apps/official-journal-web/src/pages/innskraning.tsx:40]: remove

**Review by jonbjarnio — APPROVED** [review]
Very nice 🥇

**Pull request merged**